src/DateTimeFormatter.php

```diff
@@ -78,7 +78,7 @@ public function formatDuration(float $seconds, string $locale = null): string
 
                     return $this->translator->trans(
                         $format[1],
-                        ['%count%' => floor($seconds / $format[2])],
+                        ['%count%' => number_format($seconds / $format[2], $precision)],
```

Comment:
```suggestion
                        ['%count%' => null === $precision ? floor($seconds / $format[2]) : number_format($seconds / $format[2], $precision)],
```

To keep the current behaviour.

Reply:
My previous logic was wrong :( I remove the precision parameter and add a private method that format the seconds into float number.